src/Microsoft.AspNet.SignalR.JS/jquery.signalR.core.js

```diff
@@ -729,86 +730,95 @@
                         }
 
                         // Check for a redirect response (which must have a ProtocolVersion of 2.0)
-                        if (res.ProtocolVersion === "2.0" && res.RedirectUrl) {
-                            if (redirects === MAX_REDIRECTS) {
-                                onFailed(signalR._.error(resources.errorRedirectionExceedsLimit), connection);
+                        if (res.ProtocolVersion === "2.0") {
+                            if (res.Error) {
+                                protocolError = signalR._.error(signalR._.format(resources.errorFromServer, res.Error));
+                                $(connection).triggerHandler(events.onError, [protocolError]);
+                                deferred.reject(protocolError);
                                 return;
                             }
+                            else if (res.RedirectUrl) {
+                                if (redirects === MAX_REDIRECTS) {
```

Comment:
nit: Don't think this could happen but it should be greater than or equal to

Reply:
We always increment by 1 so it shouldn't happen. Also this PR is just moving that code around, the code is as it was before.